code/modules/mob/new_player/new_player.dm

```diff
@@ -190,10 +190,10 @@
 
 				observer.timeofdeath = world.time
 
-				var/datum/species/species = GLOB.all_species[client.prefs.species] || GLOB.all_species[DEFAULT_SPECIES]
+				var/datum/species/species = GLOB.roundstart_species[client.prefs.species] || GLOB.roundstart_species[DEFAULT_SPECIES]
```

Comment:
leave as all_species, like the others you want to look in all species, only when choosing you should show roundstart_species as option or when defaulting it.